File: Factory/Java/LogisticsFactory/src/LogisticsFactory/LogisticsFactory.java
package LogisticsFactory;

public class LogisticsFactory {
	private static RoadLogistics road = new RoadLogistics();
	private static SeaLogistics sea = new SeaLogistics();
	private static AirLogistics air = new AirLogistics();
	
	public static Transport createTransport(String type, String whatToDeliver, Coordinate from, Coordinate to)
	{
		if(type.equalsIgnoreCase("sea"))
			return sea.createTransport(whatToDeliver, from, to);
		else if(type.equalsIgnoreCase("road"))
			return road.createTransport(whatToDeliver, from, to);
		else if(type.equalsIgnoreCase("air"))
			return air.createTransport(whatToDeliver, from, to);
		
		return new UnknownTransport(whatToDeliver, null);
	}
}

File: Factory/C#/Logistics/Factory/LogisticsFactory.cs
namespace Factory
{
    public class LogisticsFactory
    {
        private static readonly RoadLogistics Road = new RoadLogistics();
        private static readonly SeaLogistics Sea = new SeaLogistics();
        private static readonly AirLogistics Air = new AirLogistics();

        public static Transport CreateTransport(string type, string whatToDeliver, Coordinate from, Coordinate to)
        {
            if (type.Equals("sea"))
                return Sea.CreateTransport(whatToDeliver, from, to);
            else if (type.Equals("road"))
                return Road.CreateTransport(whatToDeliver, from, to);
            else if(type.Equals("air"))
                return Air.CreateTransport(whatToDeliver, from, to);

            return new UnknownTransport(whatToDeliver, null);
        }
    }
}